tpot/operators/classifiers/base.py

```diff
@@ -57,8 +56,22 @@ def _train_and_predict(self, input_matrix, operator_args):
         clf.fit(self.training_features, self.training_classes)
 
         all_features = np.delete(input_matrix, non_feature_columns, axis=1)
-        input_matrix[:, GUESS_COL] = clf.predict(all_features)
+        if self.clf_eval_func == 'decision_function':
+            if 'decision_function' in dir(clf):
+                input_matrix = np.insert(input_matrix, GUESS_COL, clf.decision_function(all_features).T, axis=1)
```

Comment:
Sorry about that. To clarify: certain scoring functions need an array of labels of size (n_samples,) as well as a matrix of prediction probabilities (or in some cases, decision function values) of size (n_samples, n_classes). This means using either the predict_proba() or decision_function() methods (and putting filler in if the classifier doesn't implement the function necessary). Originally I had been pickling these values into strings and putting them in the single guess column, but that was assuming an underlying dataframe. Rather than gutting the existing numpy matrix code, I figured it'd be easier to insert the (n_samples, n_classes) into the matrix at GUESS_COL and just try to keep track of all the added columns.